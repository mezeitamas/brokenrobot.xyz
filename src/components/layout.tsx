import React from 'react';
import type { FunctionComponent, ReactElement, PropsWithChildren } from 'react';

import { FiRss, FiGithub, FiLinkedin } from 'react-icons/fi';

import { useSiteMetadata } from '../hooks/use-site-metadata';

import { ExternalLink } from './external-link';
import { InternalLink } from './internal-link';

const Layout: FunctionComponent<PropsWithChildren> = ({ children }): ReactElement => {
    const siteMetadata = useSiteMetadata();

    return (
        <div className="container mx-auto bg-white px-4">
            <header className="mb-12 mt-6 flex flex-col gap-4 leading-8">
                <nav className="flex flex-col justify-between gap-4 whitespace-nowrap sm:flex-row">
                    <div className="text-2xl font-bold text-gray-900">
                        <InternalLink to="/">{siteMetadata.title}</InternalLink>
                    </div>

                    <div className="flex flex-nowrap gap-6 text-gray-600 sm:gap-10">
                        <InternalLink to="/">Home</InternalLink>
                        <InternalLink to="/blog">Blog</InternalLink>
                        <InternalLink to="/about-me">About me</InternalLink>
                    </div>
                </nav>

                <hr />
            </header>

            <main className="prose max-w-none">{children}</main>

            <footer className="mb-6 mt-16 flex flex-col gap-4">
                <hr />

                <div className="flex flex-col justify-between gap-4 whitespace-nowrap text-gray-700 sm:flex-row">
                    <div className="flex gap-6 sm:gap-10">
                        <div className="flex items-center gap-2">
                            <FiGithub size="18" />
                            <ExternalLink href={siteMetadata.author.social.github}>GitHub</ExternalLink>
                        </div>

                        <div className="flex items-center gap-2">
                            <FiLinkedin size="18" />
                            <ExternalLink href={siteMetadata.author.social.linkedin}>LinkedIn</ExternalLink>
                        </div>
                    </div>

                    <div className="flex items-center gap-2">
                        <FiRss size="18" />
                        <ExternalLink href="/rss.xml">RSS Feed</ExternalLink>
                    </div>
                </div>

                <div className="text-sm text-gray-400">
                    &copy; {new Date().getFullYear()} {siteMetadata.author.name}. All rights reserved.
                </div>
            </footer>
        </div>
    );
};

export { Layout };
