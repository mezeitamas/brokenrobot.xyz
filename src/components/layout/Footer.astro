---
import { GithubIcon, LinkedinIcon, GlobeIcon, TwitterIcon, RssIcon } from 'astro-feather';

import ExternalLink from '@components/links/ExternalLink.astro';
import InternalLink from '@components/links/InternalLink.astro';

type Props = {
    githubLink: string;
    linkedinLink: string;
    mastodonLink: string;
    twitterLink: string;
    authorName: string;
};

const { githubLink, linkedinLink, mastodonLink, twitterLink, authorName } = Astro.props;
const today = new Date();
---

<footer class="mb-6 mt-12 flex flex-col gap-4 text-gray-600">
    <div class="flex flex-col justify-between gap-6 whitespace-nowrap sm:flex-row sm:gap-10">
        <div class="flex flex-col gap-4 sm:flex-row">
            <div class="flex flex-row gap-4">
                <div class="flex items-center gap-2">
                    <GithubIcon />
                    <ExternalLink href={githubLink}>GitHub</ExternalLink>
                </div>

                <div class="flex items-center gap-2">
                    <LinkedinIcon />
                    <ExternalLink href={linkedinLink}>LinkedIn</ExternalLink>
                </div>
            </div>

            <div class="flex flex-row gap-4">
                <div class="flex items-center gap-2">
                    <GlobeIcon />
                    <ExternalLink
                        href={mastodonLink}
                        rel="me"
                    >
                        Mastodon
                    </ExternalLink>
                </div>

                <div class="flex items-center gap-2">
                    <TwitterIcon />
                    <ExternalLink href={twitterLink}>Twitter</ExternalLink>
                </div>
            </div>
        </div>

        <div class="flex items-center gap-2">
            <RssIcon />
            <InternalLink href="/rss.xml">RSS Feed</InternalLink>
        </div>
    </div>

    <div class="text-sm">&copy; {today.getFullYear()} {` `} {authorName}. All rights reserved.</div>
</footer>
