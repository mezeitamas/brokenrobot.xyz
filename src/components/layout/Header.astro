---
import InternalLink from '../links/InternalLink.astro';

type Props = {
    title: string;
};

const { title } = Astro.props;
---

<header class="mb-12 mt-6 flex flex-col gap-4 leading-8">
    <nav class="flex flex-col justify-between gap-4 whitespace-nowrap sm:flex-row">
        <div class="text-2xl font-bold text-gray-900">
            <InternalLink href="/">{title}</InternalLink>
        </div>

        <div class="flex flex-nowrap gap-6 text-gray-600 sm:gap-10">
            <InternalLink href="/">Home</InternalLink>
            <InternalLink href="/blog">Blog</InternalLink>
            <InternalLink href="/about">About</InternalLink>
        </div>
    </nav>

    <hr />
</header>
