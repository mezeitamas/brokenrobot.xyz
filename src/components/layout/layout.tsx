import React from 'react';
import type { FunctionComponent, ReactElement, PropsWithChildren } from 'react';

import { useSiteMetadata } from '../site-metadata/use-site-metadata';

import { Content } from './content/content';
import { Footer } from './footer/footer';
import { Header } from './header/header';

const Layout: FunctionComponent<PropsWithChildren> = ({ children }): ReactElement => {
    const { title, author } = useSiteMetadata();

    return (
        <div className="container mx-auto bg-white px-4">
            <Header title={title} />

            <Content>{children}</Content>

            <Footer
                githubLink={author.social.githubUrl}
                linkedinLink={author.social.linkedinUrl}
                twitterLink={author.social.twitterUrl}
                authorName={author.name}
            />
        </div>
    );
};

export { Layout };
