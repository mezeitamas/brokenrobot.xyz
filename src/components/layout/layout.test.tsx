import React from 'react';
import type { PropsWithChildren } from 'react';

import { create } from 'react-test-renderer';

import type { SiteMetadata } from '../site-metadata/site-metadata.types';

import { Layout } from './layout';

jest.mock('../site-metadata/use-site-metadata', () => {
    return {
        useSiteMetadata: jest.fn().mockImplementation((): SiteMetadata => {
            return {
                title: 'Broken Robot',
                description:
                    'Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world.',
                siteUrl: 'https://www.brokenrobot.xyz',
                author: {
                    name: 'Tamas Mezei',
                    social: {
                        githubUrl: 'https://github.com/mezeitamas',
                        linkedinUrl: 'https://www.linkedin.com/in/mezeitamas/',
                        mastodonUrl: 'https://mastodon.social/@brokenrobot_xyz',
                        mastodonUsername: '@brokenrobot_xyz@mastodon.social',
                        twitterUrl: 'https://twitter.com/brokenrobot_xyz',
                        twitterUsername: '@brokenrobot_xyz'
                    }
                }
            };
        })
    };
});

describe('<Layout />', () => {
    test('should render', () => {
        // Given
        const props: PropsWithChildren = {
            children: <div>Content</div>
        };

        // When
        const component = create(<Layout {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
