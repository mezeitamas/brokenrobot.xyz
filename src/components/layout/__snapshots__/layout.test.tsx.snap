// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render 1`] = `
<div
  className="mx-auto max-w-screen-xl px-4 md:px-6 lg:px-8"
>
  <header
    className="mb-12 mt-6 flex flex-col gap-4 leading-8"
  >
    <nav
      className="flex flex-col justify-between gap-4 whitespace-nowrap sm:flex-row"
    >
      <div
        className="text-2xl font-bold text-gray-900"
      >
        <span
          className="hover:underline"
        >
          <a
            href="/"
          >
            Broken Robot
          </a>
        </span>
      </div>
      <div
        className="flex flex-nowrap gap-6 text-gray-600 sm:gap-10"
      >
        <span
          className="hover:underline"
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
        <span
          className="hover:underline"
        >
          <a
            href="/blog"
          >
            Blog
          </a>
        </span>
        <span
          className="hover:underline"
        >
          <a
            href="/about-me"
          >
            About me
          </a>
        </span>
      </div>
    </nav>
    <hr />
  </header>
  <main
    className="prose max-w-none"
  >
    <div>
      Content
    </div>
  </main>
  <footer
    className="mb-6 mt-16 flex flex-col gap-4 text-gray-700"
  >
    <hr />
    <div
      className="flex flex-col justify-between gap-6 whitespace-nowrap sm:flex-row sm:gap-10"
    >
      <div
        className="flex flex-col gap-4 sm:flex-row"
      >
        <div
          className="flex flex-row gap-4"
        >
          <div
            className="flex items-center gap-2"
          >
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
              />
            </svg>
            <span
              className="inline-flex items-center gap-1 hover:underline"
            >
              <a
                href="https://github.com/mezeitamas"
                target="_blank"
              >
                GitHub
              </a>
              <svg
                fill="none"
                height="14"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <polyline
                  points="15 3 21 3 21 9"
                />
                <line
                  x1="10"
                  x2="21"
                  y1="14"
                  y2="3"
                />
              </svg>
            </span>
          </div>
          <div
            className="flex items-center gap-2"
          >
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
              />
              <rect
                height="12"
                width="4"
                x="2"
                y="9"
              />
              <circle
                cx="4"
                cy="4"
                r="2"
              />
            </svg>
            <span
              className="inline-flex items-center gap-1 hover:underline"
            >
              <a
                href="https://www.linkedin.com/in/mezeitamas/"
                target="_blank"
              >
                LinkedIn
              </a>
              <svg
                fill="none"
                height="14"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <polyline
                  points="15 3 21 3 21 9"
                />
                <line
                  x1="10"
                  x2="21"
                  y1="14"
                  y2="3"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div
            className="flex items-center gap-2"
          >
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M18.648 15.254c-1.816 1.763 -6.648 1.626 -6.648 1.626a18.262 18.262 0 0 1 -3.288 -.256c1.127 1.985 4.12 2.81 8.982 2.475c-1.945 2.013 -13.598 5.257 -13.668 -7.636l-.026 -1.154c0 -3.036 .023 -4.115 1.352 -5.633c1.671 -1.91 6.648 -1.666 6.648 -1.666s4.977 -.243 6.648 1.667c1.329 1.518 1.352 2.597 1.352 5.633s-.456 4.074 -1.352 4.944z"
              />
              <path
                d="M12 11.204v-2.926c0 -1.258 -.895 -2.278 -2 -2.278s-2 1.02 -2 2.278v4.722m4 -4.722c0 -1.258 .895 -2.278 2 -2.278s2 1.02 2 2.278v4.722"
              />
            </svg>
            <span
              className="inline-flex items-center gap-1 hover:underline"
            >
              <a
                href="https://mastodon.social/@brokenrobot_xyz"
                rel="me"
                target="_blank"
              >
                Mastodon
              </a>
              <svg
                fill="none"
                height="14"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <polyline
                  points="15 3 21 3 21 9"
                />
                <line
                  x1="10"
                  x2="21"
                  y1="14"
                  y2="3"
                />
              </svg>
            </span>
          </div>
          <div
            className="flex items-center gap-2"
          >
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
              />
            </svg>
            <span
              className="inline-flex items-center gap-1 hover:underline"
            >
              <a
                href="https://twitter.com/brokenrobot_xyz"
                target="_blank"
              >
                Twitter
              </a>
              <svg
                fill="none"
                height="14"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                />
                <polyline
                  points="15 3 21 3 21 9"
                />
                <line
                  x1="10"
                  x2="21"
                  y1="14"
                  y2="3"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="flex items-center gap-2"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 11a9 9 0 0 1 9 9"
          />
          <path
            d="M4 4a16 16 0 0 1 16 16"
          />
          <circle
            cx="5"
            cy="19"
            r="1"
          />
        </svg>
        <span
          className="hover:underline"
        >
          <a
            href="/rss.xml"
          >
            RSS Feed
          </a>
        </span>
      </div>
    </div>
    <div
      className="text-sm"
    >
      © 2023 
      Tamas Mezei
      . All rights reserved.
    </div>
  </footer>
</div>
`;
